lua/lsp/init.lua

```diff
@@ -128,22 +128,20 @@ function M.common_on_attach(client, bufnr)
 end
 
 function M.setup(lang)
-  local lsp_utils = require "lsp.utils"
-  local lsp = lvim.lang[lang].lsp
-  if lsp_utils.is_client_active(lsp.provider) then
-    return
-  end
-
   local overrides = lvim.lsp.override
   if type(overrides) == "table" then
     if vim.tbl_contains(overrides, lang) then
       return
     end
   end
 
-  if lsp.provider ~= nil and lsp.provider ~= "" then
-    local lspconfig = require "lspconfig"
-    lspconfig[lsp.provider].setup(lsp.setup)
+  -- initialize language server for each provider
+  local lsp_utils = require "lsp.utils"
+  local providers = lvim.lang[lang].lsp.providers
```

Comment:
what happens if `lvim.lang[lang]` is nil?
for example for a lang that we didn't define in `default-config.lua`

Reply:
Managed to test this by deleting an entry from `default-config.lua`. It does indeed throw an error.
The one-line change I pushed resolves the issue by adding a fallback `{}` if that value is empty.